precompiles/utils/macro/src/precompile/expand.rs

```diff
@@ -416,8 +416,11 @@ impl Precompile {
 						)
 					}
 
-					fn is_precompile(&self, address: H160) -> bool {
-						<#impl_type>::#discriminant_fn(address).is_some()
+					fn is_precompile(&self, address: H160, _gas: u64) -> ::fp_evm::IsPrecompileResult {
+						fp_evm::IsPrecompileResult::Answer {
+							is_precompile: <#impl_type>::#discriminant_fn(address).is_some(),
```

Comment:
We should probably change the discriminant function signature to check and report costs, as our Assets ERC20 precompile does a storage read to know if an asset exists.

Reply:
sorry @nanocryk this was moved to https://github.com/PureStake/moonbeam/pull/2312

please review there